FlowCrypt/Core/Core Services/DataBaseService.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  DataBaseService.swift
+//  FlowCrypt
+//
+//  Created by Anton Kharchevskyi on 8/22/19.
+//  Copyright © 2019 FlowCrypt Limited. All rights reserved.
+//
+
+import Foundation
+import Promises
+import RealmSwift
+
+enum DataBaseError {
+    case failedToSave
+}
+
+protocol DataBaseService {
+    func save(message data: Data, isEmail: Bool) -> Promise<CoreRes.ParseDecryptMsg>
+}
+
+struct RealmDataBaseService: DataBaseService {
+    static let shared = RealmDataBaseService()
+
+    func save(message data: Data, isEmail: Bool) -> Promise<CoreRes.ParseDecryptMsg> {
+        return Promise { resolve, reject in
```

Comment:
Same comment as above, plus:
1) This does not need to return a Promise. There's no async functionality here. Promises are useful in situations where we need to handle a callback, with an escaping closure, and want to avoid callback hell. No callbacks here -> no need for Promises.
2) Don't hide decryption in a function that has nothing to do with it (save)
3) Don't call it `save` if there is nothing being saved

I recommend to delete this file completely, and put:
```Swift
            let realm = try Realm()
            let keys = PrvKeyInfo.from(realm: realm.objects(KeyInfo.self))
            let decrypted = try Core.parseDecryptMsg(
              encrypted: data,
              keys: keys,
              msgPwd: nil,
              isEmail: isEmail
           )
```
Into `MsgViewController` where it belongs.

There should not be a need for do/catch either, because you can `try` things inside a promise `.then` (in MsgViewController), and any thrown errors should then surface to `.catch`.